pkg/experiment/sensitivity/workload_factory.go

```diff
@@ -0,0 +1,276 @@
+	factory ExecutorFactory,
+	hpIsolation isolation.Decorator,
+	l1Isolation isolation.Decorator,
+	l3Isolation isolation.Decorator) WorkloadFactory {
+	return WorkloadFactory{
+		executorFactory: factory,
+		hpIsolation:     hpIsolation,
+		l1Isolation:     l1Isolation,
+		l3Isolation:     l3Isolation,
+	}
+}
+
+// BuildDefaultHighPriorityLauncher builds High Priority workload launcher with predefined isolation.
+func (factory *WorkloadFactory) BuildDefaultHighPriorityLauncher(
+	workloadName string, tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.BuildHighPriorityLauncherWithIsolation(workloadName, factory.hpIsolation, tags)
+}
+
+// BuildHighPriorityLauncherWithIsolation builds High Priority launcher with provided isolation.
+func (factory *WorkloadFactory) BuildHighPriorityLauncherWithIsolation(
+	workloadName string,
+	decorators isolation.Decorator,
+	tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.createHighPriorityWorkload(workloadName, decorators, tags)
+}
+
+// BuildDefaultBestEffortLauncher builds Best Effort workload launcher with predefined isolation.
+func (factory *WorkloadFactory) BuildDefaultBestEffortLauncher(
+	workloadName string,
+	tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.BuildBestEffortLauncherWithIsolation(workloadName, factory.getDefaultBestEffortIsolation(workloadName), tags)
+}
+
+// BuildBestEffortLauncherWithIsolation builds Best Effort launcher with provided isolation.
+func (factory *WorkloadFactory) BuildBestEffortLauncherWithIsolation(
+	workloadName string,
+	isolation isolation.Decorator,
+	tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.createBestEffortWorkload(workloadName, isolation, tags)
+}
+
+func (factory *WorkloadFactory) createHighPriorityWorkload(
+	name string,
+	isolation isolation.Decorator,
+	tags snap.Tags) (executor.Launcher, error) {
+
+	exec, err := factory.executorFactory.BuildBestEffortExecutor(isolation, isolation)
+	if err != nil {
+		return nil, err
+	}
+
+	switch name {
+	case Memcached:
+		return executor.NewServiceLauncher(memcached.New(exec, memcached.DefaultMemcachedConfig())), nil
+	case Specjbb:
+		return executor.NewServiceLauncher(specjbb.NewBackend(exec, specjbb.DefaultSPECjbbBackendConfig())), nil
+	default:
+		return nil, errors.Errorf("unknown high priority task %q", name)
+	}
+}
+
+func (factory *WorkloadFactory) createBestEffortWorkload(
+	name string,
+	isolation isolation.Decorator,
+	tags snap.Tags) (executor.Launcher, error) {
+	// Zero-value sensitivity.LauncherSessionPair represents baseline.
+	if name == NoneAggressorID {
+		return nil, nil
+	}
+
+	var workload executor.Launcher
+	additionalDecorators := factory.getBestEffortAdditionalDecorators(name)
+	exec, err := factory.executorFactory.BuildBestEffortExecutor(isolation, additionalDecorators)
+	if err != nil {
+		return nil, err
+	}
+
+	// Best Effort workloads.
+	switch name {
```

Comment:
It might be good call for the next time. Currently, we have to change config in Caffe to supply tags.